### DISCUSSION TITLE
VScode+fortls: #include not recongnized

### DISCUSSION THREAD

**Discussion by tommy-engels** [fortran-lang/fortls#353] [Q&A] [answered]
Dear all,

I have a problem that seems to be benign, yet I could not figure out the solution myself. 

My FORTRAN code uses 

`#include "file.f90" `

directives, rather than the fortran-build-in

`include("file.f90")`

because the latter does not feature preprocessor directives.
However, if I open the code in visual studio code, I get many "problem" messages that are related to fortls not understanding these include statements.

In particular, I have a module:

```
module hello
...
integer :: global_variable
...
#include "file.f90"
...
end module
```

and if I edit the file `file.f90`, VS studio complains that `global_variable` is not defined in the file. 

I realize that `fortls` should be capable of dealing with this, but I don't quite know what could go wrong and where to look. I would be grateful for any pointer!

**Comment by nsh-envs** [comment]
I don't know if this was ever addressed. I at least have the same problem still, and i think this would be a nice feature to add. 

Did you end up fixing it yourself?

**Comment by tommy-engels** [comment]
I did not manage to find a fix for it, nor has it caught the developers attention :)

**Comment by nsh-envs** [comment]
Okay, thanks for letting me know. 

It seems there could be a fix in the file `fortls/langserver.py`

```Python
    def serve_document_symbols(self, request: dict):
        ...
        # Add scopes to outline view
        test_output = []
        for scope in file_obj.ast.get_scopes():
            if (
                not scope.name  # Skip empty strings
                or scope.name.startswith("#")  # Skip comments                      <- Here
                or scope.get_type() == SELECT_TYPE_ID  # Skip select types
            ):
                continue
        ...
```

I don't know if there is other places it would have to change, or if the preprocessor can be activated in a different way. This would suggest compiler directives is not supported or handled somewhere else?

**Comment by tommy-engels** [comment]
I realized that this thread is in the "discussions" and not posted as an issue. maybe that is why it went unnoticed? 
I myself am not capable of helping in fixing this issue, I am afraid

**Comment by nsh-envs** [comment, marked as answer]
Yes, maybe marked as enhancement or bug as it should be compatible.

**Comment by tommy-engels** [comment]
I created the issue here:
https://github.com/fortran-lang/fortls/issues/495
